ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,603 @@
+            node.tooltipPos = chart.inverted ? [
+                (chart.plotSizeY || 0) - y - height / 2,
+                (chart.plotSizeX || 0) - x - width / 2
+            ] : [
+                x + width / 2,
+                y + height / 2
+            ];
+
+            if (node.shapeType === 'circle') {
+                node.shapeArgs = {
+                    x: x + width / 2,
+                    y: y,
+                    r: height / 2,
+                    width: width,
+                    height,
+                    display: node.hasShape() ? '' : 'none'
+                };
+
+                node.dlBox = {
+                    x: x + width / 2,
+                    y: y,
+                    height: 0,
+                    width: 0
+                };
+
+            } else {
+                node.shapeArgs = {
+                    x,
+                    y,
+                    width,
+                    height,
+                    display: node.hasShape() ? '' : 'none'
+                };
+            }
+
+        } else {
+            node.dlOptions = {
+                enabled: false
+            };
+        }
+    }
+
+    /* eslint-enable valid-jsdoc */
+}
+
+/* *
+ *
+ *  Prototype Properties
+ *
+ * */
+
+interface ArcDiagramSeries {
+    orderNodes: boolean;
+    pointClass: typeof ArcDiagramPoint;
+}
+extend(ArcDiagramSeries.prototype, {
+    orderNodes: false
+});
+
+/* *
+ *
+ *  Registry
+ *
+ * */
+
+declare module '../../Core/Series/SeriesType' {
+    interface SeriesTypeRegistry {
+        arcdiagram: typeof ArcDiagramSeries;
+    }
+}
+ArcDiagramSeries.prototype.pointClass = ArcDiagramPoint;
+SeriesRegistry.registerSeriesType('arcdiagram', ArcDiagramSeries);
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default ArcDiagramSeries;
+
+/* *
+ *
+ *  API Options
+ *
+ * */
+
+/**
+ * An `arcdiagram` series. If the [type](#series.arcdiagram.type)
+ * option is not specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.arcdiagram
+ * @exclude   dataSorting
+ * @product   highcharts
+ * @requires  modules/sankey
+ * @requires  modules/arc-diagram
+ * @apioption series.arcdiagram
+ */
+
+/**
+ * A collection of options for the individual nodes. The nodes in an arc
+ * diagram are auto-generated instances of `Highcharts.Point`, but options can
+ * be applied here and linked by the `id`.
+ *
+ * @extends   series.arcdiagram.nodes
+ * @type      {Array<*>}
+ * @product   highcharts
+ * @excluding offset
+ * @apioption series.arcdiagram.nodes
+ */
+
+/**
+ * An array of data points for the series. For the `arcdiagram` series
+ * type, points can be given in the following way:
+ *
+ * An array of objects with named values. The following snippet shows only a
+ * few settings, see the complete options set below. If the total number of data
+ * points exceeds the series' [turboThreshold](#series.area.turboThreshold),
+ * this option is not available.
```

Comment:
Since it's the only option to configure the data, then shouldn't be `turboThreshold` disabled? 🤔 

It seems it's not the only series with this bug. other types: dependency wheel, flags, networkgraph, organization, sankey. Perhaps default `keys` should be set to `['from', 'to', 'weight']`? This is the way `MapSeries` and `Wordcloud` solve this issue (provide two default ways to configure the data).